src/test/java/org/checkstyle/autofix/recipe/UpperEllRecipeTest.java

```diff
@@ -1,40 +1,71 @@
 
-import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
 import static org.openrewrite.java.Assertions.java;
 
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
 import org.junit.jupiter.api.Test;
 import org.openrewrite.test.RecipeSpec;
 import org.openrewrite.test.RewriteTest;
 
 public class UpperEllRecipeTest implements RewriteTest {
 
+    private static final String BASE_TEST_RESOURCES_PATH = "src/test/resources/org"
+            + "/checkstyle/autofix/recipe/";
+
     @Override
     public void defaults(RecipeSpec spec) {
         spec.recipe(new UpperEllRecipe());
     }
 
     @Test
-    void fixesLowercase() {
-        rewriteRun(
-
-            java(
-                "class Test {\n"
-                        + "    int value1 = 123l;\n"
-                        + "    long value2 = 0x123l;\n"
-                        + "    long value3 = 0123l;\n"
-                        + "    long value4 = 0b101l;\n"
-                        + "    String value5 = null;\n"
-                        + "}\n",
-                "class Test {\n"
-                        + "    int value1 = 123L;\n"
-                        + "    long value2 = 0x123L;\n"
-                        + "    long value3 = 0123L;\n"
-                        + "    long value4 = 0b101L;\n"
-                        + "    String value5 = null;\n"
-                        + "}\n"
-            )
-        );
-        assertTrue(true, "Test completed successfully");
+    void longLiteralTest() throws IOException {
+        final String beforeCode = Files.readString(Paths.get(BASE_TEST_RESOURCES_PATH
+                + "upperell/longliteral/InputLongLiteral.txt"));
+
+        final String afterCode = Files.readString(Paths.get(BASE_TEST_RESOURCES_PATH
+                + "upperell/longliteral/OutputLongLiteral.txt"));
+
+        assertDoesNotThrow(() -> rewriteRun(java(beforeCode, afterCode)));
+    }
+
+    @Test
+    void hexOctalLiteralTest() throws IOException {
+
+        final String beforeCode = Files.readString(Paths.get(BASE_TEST_RESOURCES_PATH
+                + "upperell/hexoctalliteral/InputHexOctalLiteral.txt"));
+
+        final String afterCode = Files.readString(Paths.get(BASE_TEST_RESOURCES_PATH
+                + "upperell/hexoctalliteral/OutputHexOctalLiteral.txt"));
+
+        assertDoesNotThrow(() -> rewriteRun(java(beforeCode, afterCode)));
     }
+
+    @Test
+    void complexLongLiterals() throws IOException {
+
+        final String beforeCode = Files.readString(Paths.get(BASE_TEST_RESOURCES_PATH
+                + "upperell/complexlongliterals/InputComplexLongLiterals.txt"));
+
+        final String afterCode = Files.readString(Paths.get(BASE_TEST_RESOURCES_PATH
+                + "upperell/complexlongliterals/OutputComplexLongLiterals.txt"));
+
+        assertDoesNotThrow(() -> rewriteRun(java(beforeCode, afterCode)));
```

Comment:
this logic is duplicated all around, let's create a helper test class, move it there and make tests look something like:
```java
@Test
void complexLongLiterals() {
  testRecipe("ComplexLongLiterals");
}
```